lib/postgrex/connection.ex

```diff
@@ -167,16 +134,11 @@ defmodule Postgrex.Connection do
 
     * `:timeout` - Call timeout (default: `#{@timeout}`)
   """
-  @spec listen(pid, String.t, Keyword.t) :: {:ok, reference} | {:error, Postgrex.Error.t}
+  @spec listen(pid, String.t, Keyword.t) :: {:ok, reference}
```

Comment:
Merge it as is and create a new issue so we fix it during or after the split if you find it worth. No reason putting a lot of effort into this if the solution is going to change soon anyway.